packages/dear-imgui/cpp/include/widget/map_view.h

```diff
@@ -12,13 +12,11 @@ class MapView final : public StyledWidget {
 
 public:
     static std::unique_ptr<MapView> makeWidget(const json& widgetDef, std::optional<WidgetStyle> maybeStyle, ReactImgui* view) {
-        if (widgetDef.is_object() && widgetDef.contains("id") && widgetDef["id"].is_number_integer()) {
-            auto id = widgetDef["id"].template get<int>();
+        auto id = widgetDef["id"].template get<int>();
 
-            return std::make_unique<MapView>(view, id, maybeStyle);
-        }
+        return std::make_unique<MapView>(view, id, maybeStyle);
 
-        throw std::invalid_argument("Invalid JSON data");
+        // throw std::invalid_argument("Invalid JSON data");
```

Comment:
Instead of validating elementDef everywhere, I validate that it is an object and that it contains the id integer property one time only in ReactImgui::CreateElement. It's a calculated risk I am willing to take to boost performance a little bit.